packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -62,6 +88,46 @@ class ComponentStateBucket {
 }
 
 class CurlyComponentManager {
+  prepareArgs(definition, args) {
+    validatePositionalParameters(args.named, args.positional.values, definition.ComponentClass.positionalParams);
+
+    if (definition.args) {
+      // args (aka 'oldArgs') may be undefined or simply be empty args, so
+      // we need to fall back to an empty array or object.
+      let newNamed = args && args.named && args.named.map || {};
+      let newPositional = args && args.positional && args.positional.values || [];
+
+      let oldNamed = definition.args.named.map;
+      let oldPositional = definition.args.positional.values;
+
+      // Merge positional arrays
+      let mergedPositional = new Array(Math.max(oldPositional.length, newPositional.length));
+      mergedPositional.splice(0, oldPositional.length, ...oldPositional);
+      mergedPositional.splice(0, newPositional.length, ...newPositional);
+
+      // Merge named maps
+      let mergedNamed = assign({}, oldNamed, newNamed);
+
+      // THOUGHT: It might be nice to have a static method on EvaluatedArgs that
+      // can merge two sets of args for us.
+      let mergedArgs = EvaluatedArgs.create({
+        named: EvaluatedNamedArgs.create({
+          map: mergedNamed
+        }),
+        positional: EvaluatedPositionalArgs.create({
+          values: mergedPositional
+        })
+      }).withInternal();
```

Comment:
Yeah, that whole thing needs rework. What I mean is that you only need to call `withInternal` if you are 1) possibly dealing with a shared object 2) you are about to put stuff into the dictionary.

For you case, you can just do:

```js
 let mergedArgs = EvaluatedArgs.create({ ... });
mergedArgs.internal = originalArgs.internal;
```